### PR TITLE
publish: sortable sidebar

### PR DESCRIPTION
**Pull request by matildepark**
Adds sidebar, four options to sort its items.

image

### CODE REVIEW

**Comment by vvisigoth** [comment]
ping @loganallenc

File: pkg/interface/publish/src/js/components/lib/sidebar.js [hunk 1]
```diff
@@ -6,62 +6,151 @@ export class Sidebar extends Component {
...
+  componentDidUpdate(prevProps, prevState) {
+    if ((prevState.sort !== this.state.sort) || (prevProps !== this.props)) {
+      this.sort();
+    }
+  }
 
-    let hiddenClasses = true;
+  sort() {
+    let { props, state } = this;
+    let notebooks = new Map();
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/sidebar.js]: sanity check these to make sure a[1] and b[1] exist before checking properties within them. program defensively!

File: pkg/interface/publish/src/js/components/lib/sidebar.js [hunk 2]
```diff
@@ -6,62 +6,151 @@ export class Sidebar extends Component {
...
+  }
 
-    if (this.props.popout) {
+  sortChange(event) {
+    this.setState({sort: event.target.value});
+  }
+
+  render() {
+    const { props, state } = this;
+    let activeClasses = (props.active === "sidebar") ? " " : "dn-s ";
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/sidebar.js]: yay looks goood

File: pkg/interface/publish/src/js/components/lib/sidebar.js [hunk 3]
```diff
@@ -6,62 +6,151 @@ export class Sidebar extends Component {
...
 
-  render() {
-    const { props, state } = this;
+  componentDidMount() {
+    this.sort();
+  }
 
-    let activeClasses = (this.props.active === "sidebar") ? " " : "dn-s ";
+  componentDidUpdate(prevProps, prevState) {
+    if ((prevState.sort !== this.state.sort) || (prevProps !== this.props)) {
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/sidebar.js]: smart and good

File: pkg/interface/publish/src/js/components/lib/sidebar.js [hunk 4]
```diff
@@ -6,62 +6,151 @@ export class Sidebar extends Component {
...
-        <a className="db dn-m dn-l dn-xl f8 pb3 pl3" href="/">⟵ Landscape</a>
-        <div className="w-100 pa4">
-          <Link
-          to="/~publish/new"
-          className="green2 mr4 f9">
-          New Notebook
+      <div
+        className={
+          "bn br-m br-l br-xl b--gray4 b--gray2-d lh-copy h-100 " +
+          "flex-shrink-0 mw-300-ns pt3 pt0-m pt0-l pt0-xl relative " +
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/sidebar.js]: yes perfect

File: pkg/interface/publish/src/js/components/lib/notebook-item.js [hunk 5]
```diff
@@ -1,13 +1,32 @@
...
+
     return (
-      <div>
-        
-      </div>
-    )
+      <Link
+      to={"/~publish/notebook/" + props.path}>
+        <div className={"w-100 v-mid f9 pl4 bb " + selectedClass}>
+        <p className="f9 pt1">{props.title}</p>
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/notebook-item.js]: indent

File: pkg/interface/publish/src/js/components/lib/notebook-item.js [hunk 6]
```diff
@@ -1,13 +1,32 @@
...
+        <p className="f9 mono gray2">by {props.author}</p>
+        <p className="f9 pb1">
+        {postCount}
+        <span className="green2 ml3">
+          {unread}
+        </span>
+        </p></div>
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/notebook-item.js]: linebreak between </p> and </div>

File: pkg/interface/publish/src/js/components/lib/notebook-item.js [hunk 7]
```diff
@@ -1,13 +1,32 @@
...
-      </div>
-    )
+      <Link
+      to={"/~publish/notebook/" + props.path}>
+        <div className={"w-100 v-mid f9 pl4 bb " + selectedClass}>
+        <p className="f9 pt1">{props.title}</p>
+        <p className="f9 mono gray2">by {props.author}</p>
+        <p className="f9 pb1">
+        {postCount}
+        <span className="green2 ml3">
```
Comment (loganallenc) [inline pkg/interface/publish/src/js/components/lib/notebook-item.js]: indent again within the p tag

**Pull request merged**